src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -252,6 +252,22 @@ void IExtensionConfigProvider.Initialize(ExtensionConfigContext context)
             rule.BindToInput<IDurableEntityClient>(this.GetClient);
             rule.BindToInput<IDurableClient>(this.GetClient);
 
+            // We add a binding rule to support the now deprecated `orchestrationClient` binding
+            // A cleaner solution would be to have the prior rule have an OR-style selector between
+            // DurableClientAttribute and OrchestrationClientAttribute, but it's unclear if that's
+            // possible (for now).
+#pragma warning disable CS0618 // Type or member is obsolete
+            var backwardsCompRule = context.AddBindingRule<OrchestrationClientAttribute>()

```

Comment:
Why is this needed? Shouldn't this work without these changes with inheritance?

Reply:
As mentioned on another issue, it sounds like you already tried removing this, but it didn't work.

Reply:
@amdeel @ConnorMcMahon 
Yup! This code is only here because inheritance didn't work by itself. 
It appears that inheritance is enough to _register_ the binding, so a `func host start` runs successfully. However, invoking the function will trigger a runtime exception, so inheritance-alone only seems to take us half of the way there.

I don't think this is the expected behaviour, since it surprises all of us. I would love to follow-up with the WebJobs folks to tell them about our usecase